src/utils/map-controls.js

```diff
@@ -270,24 +275,37 @@ export default class MapControls {
   // Default handler for the `pinchstart` event.
   _onPinchStart(event) {
     const pos = this.getCenter(event);
-    const newMapState = this.mapState.zoomStart({pos});
+    const newMapState = this.mapState.zoomStart({pos}).rotateStart({pos});
+    // hack - hammer's `rotation` field doesn't seem to produce the correct angle
+    this._state.startPinchRotation = event.rotation;
     return this.updateViewport(newMapState, NO_TRANSITION_PROPS, {isDragging: true});
   }
 
   // Default handler for the `pinch` event.
   _onPinch(event) {
-    if (!this.touchZoomRotate) {
+    if (!this.touchZoom && !this.touchRotate) {
       return false;
     }
-    const pos = this.getCenter(event);
-    const {scale} = event;
-    const newMapState = this.mapState.zoom({pos, scale});
+
+    let newMapState = this.mapState;
+    if (this.touchZoom) {
+      const {scale} = event;
+      const pos = this.getCenter(event);
+      newMapState = newMapState.zoom({pos, scale});
+    }
+    if (this.touchRotate) {
+      const {rotation} = event;
+      const {startPinchRotation} = this._state;
+      newMapState = newMapState.rotate({deltaScaleX: -(rotation - startPinchRotation) / 180});
```

Comment:
It would be so nice to be able to change pitch as well, maybe by checking the event for two fingers?